composeml/label_maker.py

```diff
@@ -232,21 +254,19 @@ def slice(self,
         if num_examples_per_instance == -1:
             num_examples_per_instance = float('inf')
 
-        for key, df in df.groupby(self.target_entity):
-            slices = self._get_slices(df=df, gap=gap, min_data=minimum_data, drop_empty=drop_empty)
+        for key, instance in df.groupby(self.target_entity):
+            slices = self._get_slices(df=instance, gap=gap, min_data=minimum_data, drop_empty=drop_empty)
 
-            for df_slice, df_metadata in slices:
-                df_metadata[self.target_entity] = key
+            for df in slices:
+                df.context.target_entity = self.target_entity
+                df.context.target_instance = key
 
                 if verbose:
-                    self.print_slice(df_metadata)
-
-                if metadata:
-                    df_slice = df_slice, df_metadata
+                    self.print_slice(df)
```

Comment:
what if `print_slice` was defined as the repr for `DataSlice` and you just do `print(df)` here?

Reply:
I like that idea. Will override `__repr__` for `print(repr(df))`.